.github/workflows/ci.yml

```diff
@@ -284,6 +297,74 @@ jobs:
           set -euo pipefail
           skaffold test -p ${SKAFFOLD_PROFILE} --build-artifacts "${{ steps.artifacts.outputs.file }}"
 
+      - name: Deploy with Skaffold
+        run: |
+          set -euo pipefail
+          skaffold deploy -p ${SKAFFOLD_PROFILE} --status-check=true --build-artifacts "${{ steps.artifacts.outputs.file }}"
+
+      - name: Verify build metadata env vars
+        run: |
+          check_env() {
+            local key="$1"
+            local expected="$2"
+            local value
+            value="$(kubectl get deploy tc -o "jsonpath={.spec.template.spec.containers[0].env[?(@.name=='${key}')].value}")"
+
+            if [ "$value" != "$expected" ]; then
+              echo "::error title=${key} mismatch::expected '${expected}', got '${value}'"
+              exit 1
+            fi
+          }
+
+          check_env "APP_VERSION" "${APP_VERSION}"
+          check_env "GIT_SHA" "${GIT_SHA}"
+          check_env "BUILD_TIME" "${BUILD_TIME}"
+
+      - name: Wait for deployments to rollout
```

Comment:
doesn't deploy already do this?

Reply:
Kept - skaffold deploy doesn't configure imagePullSecrets on the default SA. Without this, pods fail with ImagePullBackOff when pulling from GHCR.